util/stmtsummary/statement_summary.go

```diff
@@ -388,20 +410,63 @@ func (ssMap *stmtSummaryByDigestMap) SetRefreshInterval(value string, inSession
 	}
 }
 
-func (ssMap *stmtSummaryByDigestMap) RefreshInterval() int64 {
+// refreshInterval gets the refresh interval for summaries.
+func (ssMap *stmtSummaryByDigestMap) refreshInterval() int64 {
 	return atomic.LoadInt64(&ssMap.sysVars.refreshInterval)
 }
 
+// SetHistorySize sets the history size for all summaries.
+func (ssMap *stmtSummaryByDigestMap) SetHistorySize(value string, inSession bool) {
+	ssMap.sysVars.Lock()
+	if inSession {
+		ssMap.sysVars.sessionHistorySize = value
+	} else {
+		ssMap.sysVars.globalHistorySize = value
+	}
+	sessionHistorySize := ssMap.sysVars.sessionHistorySize
+	globalHistorySize := ssMap.sysVars.globalHistorySize
```

Comment:
BTW, why we need `sessionHistorySize ` and `globalHistorySize`?

Reply:
It's a little complicated.
We would like each variable to take affect in either server(session) or cluster(global) scope. If someone sets it in session scope, it will override global scope. But if he wants to unset it in session scope to make the value in global scope taking affect, he can do it by `set session xxx=''`.
So we should always save the values in both scope to get the right result.